### PR TITLE
Notify TCOs when nightly functional tests fail

### LINKED ISSUE #1235: Split nightly functional tests so TCOs can be notified upon failure
# 💡 Summary #

The Nightly tests are basically useless now for 2 reasons:

1) We have non-stop failures since Feb because AAD.  No one pays attention to constant failure.
2) No one is notified when a failure does occur.  It's hard to know when to fix something if the responsible party isn't notified of the problem.

Part of Epic #1209 

## Implementation notes ##

To fix these issues, I propose to:

1) Remove the AAD tests from Nightly and create a new issue to find a way to handle them.
2) Split the remaining tests in some manner so that one or more responsible parties are notified (maybe by email) when it fails.
3) Determine who should be notified for each product.
4) Add the notification to the workflow(s).

## Motivation and context ##

Functional tests are only useful when they are actually useful.

## Acceptance criteria ##

- [ ] All nightly tests are passing.
- [ ] At least one responsible party is notified upon failure.

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

This PR updates the nightly functional test workflow to send an email to the TCOs when the tests fails for one of their products.

Note that 1235 talks about removing AAD from the workflow because these tests had been failing for so longer.  That step was not necessary as these tests were updated to be passing.

## 💭 Motivation and context ##

Closes #1235 

## 🧪 Testing ##

Added email code, ran nightly tests manually and on a schedule, verified that emails were sent out for failures but not for passes.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] These code changes follow the ScubaGear [content style guide]
- [x] Related issues these changes resolve are linked preferably via [closing keywords]- [ ] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [ ] PR passed smoke test check.
- [ ] Feature branch has been rebased against changes from parent branch, as needed
- [ ] Resolved all merge conflicts on branch
- [ ] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: .github/workflows/test_production_function.yaml [hunk 1]
```diff
@@ -183,6 +180,10 @@ jobs:
               {
                 $m365 = $attribute
               }
+              elseif($key.ToLower() -eq "emails")
+              {
```
Comment (adhilto) [inline .github/workflows/test_production_function.yaml:184]: Style (see https://github.com/PoshCode/PowerShellPracticeAndStyle/blob/master/Style-Guide/Code-Layout-and-Formatting.md#one-true-brace-style)

File: .github/workflows/test_production_function.yaml [hunk 2]
```diff
@@ -36,6 +43,7 @@ jobs:
         id: generate-matrix
         run: |
           $params = $env:TestParams
+          echo $params.Length
```
Comment (adhilto) [inline .github/workflows/test_production_function.yaml:46]: Style (see https://github.com/PoshCode/PowerShellPracticeAndStyle/blob/d1c6ae146b1060ec4844afc393161c6237bd85a7/Style-Guide/Naming-Conventions.md#naming-conventions)
`echo` is an alias for `Write-Output`

**Review by adhilto — COMMENTED** [review]
I just have two non-critical style comments for you two consider.

File: .github/workflows/test_production_function.yaml (hunk 1 above)
Comment (james-garriss) [reply .github/workflows/test_production_function.yaml:184]: Bruh.  Can we not get hung up on things that don't matter, please?

**Review by adhilto — APPROVED** [review]

**Review by dagarwal-mitre — APPROVED** [review]
If possible add a link in the email to the Nightly Product testing workflow to the body of the email to make it easier on the TCO.

**Comment by james-garriss** [comment]
This PR will not be merged.  The notification work done by 1235 was subsumed into 1278.

**Pull request closed without merging**